packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts

```diff
@@ -88,6 +89,49 @@ export const transformPlotlyJsonToColumnProps = (jsonObj: any): IVerticalStacked
   };
 };
 
+export const transformPlotlyJsonToSankeyProps = (jsonObj: any): ISankeyChartProps => {
+  const { data, layout } = jsonObj;
+  const { link, node } = data[0];
+  const validLinks = link.value
+    .map((val: number, index: number) => ({
+      value: val,
+      source: link.source[index],
+      target: link.target[index],
+    }))
+    // eslint-disable-next-line @typescript-eslint/no-shadow
+    .filter(link => link.source !== link.target); // Filter out self-references(circular links)
+
+  const sankeyChartData = {
+    nodes: node.label.map((label: string, index: number) => ({
+      nodeId: index,
+      name: label,
+      color: node.color[index] || '#D3D3D3',
+      borderColor: node.line?.color || 'black',
+    })),
+    links: validLinks,
+  };
+
+  const width: number = layout?.width || 440;
+  const height: number = layout?.height || 220;
+  const styles: ISankeyChartProps['styles'] = {
+    root: {
+      fontSize: layout.font?.size,
+    },
+  };
+  const shouldResize: number = width + height;
```

Comment:
what does this do

Reply:
Property defined in ISankeyChartProps
/**
   * should chart resize when parent resize.
   */
  shouldResize?: number;

Reply:
but why you are adding width and height to get it.

Reply:
ok. Kshitij would know then.